### ISSUE TITLE
The design of diffusion process

### ISSUE THREAD

**Issue by Aru321** [jiang-cw/PET-diffusion#1]
Hello, I have a few questions regarding the code.

In the paper, Poisson noise is applied to the PET image to form the diffusion process. However, I couldn't locate this specific process in the project's code. It appears that the code employs traditional Gaussian noise instead of Poisson noise.

**Comment by fengjianxiong-web** [comment]
i have same question

**Comment by XYPB** [comment]
Same question here

**Comment by escwall** [comment]
the same

**Comment by PerPerZXY** [comment]
same question

**Comment by ezacngm** [comment]
evidently u didn't add any poisson noise at all, I'll appeal this to the committee,

Liar liar pants on fire!

**Comment by jiang-cw** [comment]
Hello, I am the author of this work. My apologies for the delayed response. It seems that the email may have ended up in the spam folder. I actually came to know about it through my friend.
I acknowledge that the code is incomplete. It was initially released in a rush as part of a rebuttal to reviewer. My plan was to complete it later, but this was delayed due to my need to arrange various personal matters for a year-long academic visit to the UK. I apologize for any misunderstandings and time wasted due to this. In the following, I will provide some details on the implementation of the Poisson process, and I will make it a priority to improve and update the relevant code ASAP.

Firstly, the noise imposition process in each step actually consists of three parts: 1) Converting the PET image into a sinogram via Radon transformation; 2) Applying Poisson noise on the sinogram; 3) Converting the noisy sinogram back into a noisy image.
Regarding the imposition of Poisson noise, it's important to clarify that while I am aware that Poisson noise is not additive noise but rather signal-independent noise, fully adhering to theory makes the implementation of a gradually intensifying noise diffusion process challenging. Therefore, I adopted a similar approach as used in the **Noise2Noise**  (https://github.com/NVlabs/noise2noise/blob/7355519e7bfc49e0606cca8867748e736431244b/train.py#L37), **treating Poisson noise as additive noise** applied to the image. The key code is as follows:

images_positive = images - images.min()
poisson_noise = torch.poisson(images_positive * scale) / scale
poisson_image = poisson_noise  + images - images_positive

Due to the nature of the Poisson distribution, the pixel values of the image must be non-negative.

Secondly, as I mentioned in my paper, my work on Poisson diffusion has several limitations. 1) My Poisson diffusion process is merely an engineering attempt and does not adhere to the theoretical Poisson distribution. 2) In the engineering implementation, the intensity of Poisson noise at each step was simply set to increase linearly, which theoretically should not be the case. 3) Poisson noise is not additive, so theoretically, the training loss function should be designed accordingly. However, I used a traditional loss function (predicting noise residuals). My understanding of mathematical theory is limited, so I'm not entirely sure how to address
 these limitations. These could be interesting points of research for those interested in this area.

I'm genuinely surprised and humbled that my work has garnered attention. I will make efforts to refine the code as soon as possible. (But I'm just a novice in academia, so I hope for your understanding and patience :) )
